frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -60,7 +61,8 @@ const ColumnDataEnum = {
     Sub: "Sub",
     FileName: "File Name",
     TransportResults: "Transport Results",
-};
+} as const satisfies { [k in keyof NormalizedReceiverData]: string };
+type ColumnDataTitle = (typeof ColumnDataEnum)[keyof typeof ColumnDataEnum];
```

Comment:
Hmmm I know this isn't a huge deal because a lot of things are already enumerated, but I'm wary of doing comparisons based on presentational strings like the column titles rather than using a centralized key.  

Given the interface:

```tsx
interface NormalizedReceiverData {
    Name: string;
    Service: string;
    Date: string;
    ReportId: string;
    Main: "Batch" | "Process" | "Ready";
    Sub: string;
    FileName: string;
    TransportResults: string;
}
```
(Side note: I'd still argue these should be starting with lowercase letters since they're not enums themselves but rather descriptors of object properties: `name`, `service`, `date`, `fileName` ...)

Our keys here are `Name`, `Service`, `Date`, `FileName`, et cetera.  To that end, if we're going to use these as keys, that'll make things easier because we can define titles separately:

```tsx
// renamed from ColumnDataEnum for clarity
const ColumnDataTitles = {
    name: "Name",
    service: "Service",
    date: "Date",
    reportId: "Report Id",
    main: "Main",
    sub: "Sub",
    fileName: "File Name",
    transportResults: "Transport Results",
} as const satisfies { [k in keyof NormalizedReceiverData]: string };

// will be one of "Name" | "Service" | "Date" | "FileName" | ...
type NormalizedReceiverDataKey = keyof typeof ColumnDataTitles;
```

Then we can pass these keys around as a prop:

```tsx
interface MessageReceiversColProps {
    columnKey: NormalizedReceiverDataKey;
    columnHeaderTitle: string;
    activeColumnKey?: NormalizedReceiverDataKey;
    setActiveColumnKey?: (columnKey: NormalizedReceiverDataKey) => void;
    ...
}
```

And this'd allow us to set/compare against verbatim strings because `NormalizedReceiverDataKey` would only allow for values in the union of `Name` | `Service` | `Date` | `FileName`.  So ` columnKey === "Name"` would be okay whereas `columnKey === "barf"` would not be.

The point of this would be to use the keys as the centralized identifier of columns rather than titles -- the latter of which are more prone to change.  I realize this would be a bigger change than is needed for the bugfix so I'm okay with breaking that out into a separate changeset if needed.

Reply:
I think my changes align with what you want here, it's just that i'm using the verbose keyof typeof ColumnDataTitles in all of the relevant areas instead of aliasing it to NormalizedReceiverDataKey (which should def be done for easier code reading).

Reply:
Will implement suggestions on lowercase keys, etc.

Reply:
I guess my point is more pertinent to where we're still running comparisons against the titles rather than the normalized field names instead:

- Determining class name [here](https://github.com/CDCgov/prime-reportstream/pull/8547/files#diff-9b64e4e57cfe7ca23254b7aa8cb8c29482b4f902329f5083e96c5a6be5a4657eR140)
- Assigning values [here](https://github.com/CDCgov/prime-reportstream/pull/8547/files#diff-9b64e4e57cfe7ca23254b7aa8cb8c29482b4f902329f5083e96c5a6be5a4657eR281)

Also, I know that this logic existed beforehand, but I think `formatMessages` could be simplified since we have an unnecessary iteration of `ColumnDataTitles` each invocation:

```ts
export function formatMessages(data: ReceiverData[]): NormalizedReceiverData[] {
    return data.map((receiverItem: ReceiverData): NormalizedReceiverData => {
        const fieldNames = Object.keys(ColumnDataTitles) as NormalizedReceiverKey[];

        const formattedData = fieldNames.reduce(
            (
                acc: Partial<NormalizedReceiverData>,
                fieldName: NormalizedReceiverKey
            ) => {
                let value: string | null;

                switch (fieldName) {
                    case "name":
                        value = receiverItem.receivingOrg;
                        break;
                    case "service":
                        value = receiverItem.receivingOrgSvc;
                        break;
                    case "date":
                        value = dateTimeFormatter.format(
                            new Date(receiverItem.createdAt)
                        );
                        break;
                    case "reportId":
                        value = receiverItem.reportId;
                        break;
                    case "main":
                        const { folderLocation } = parseFileLocation(
                            receiverItem.fileUrl || ""
                        );
                        value = folderLocation;
                        break;
                    case "sub":
                        const { sendingOrg } = parseFileLocation(
                            receiverItem.fileUrl || ""
                        );
                        value = sendingOrg;
                        break;
                    case "fileName":
                        const { fileName } = parseFileLocation(
                            receiverItem.fileUrl || ""
                        );
                        value = fileName;
                        break;
                    case "transportResults":
                        value = receiverItem.transportResult;
                        break;
                }

                return {
                    ...acc,
                    [fieldName]: value || "N/A",
                };
            },
            {}
        );

        return formattedData as NormalizedReceiverData;
    });
}
```

I don't think we need that initial iteration just to prepopulate all fields with "N/A" 🤔

Reply:
I'll add a ticket for this!

Reply:
https://app.zenhub.com/workspaces/web-experience-607d9d5e68b95200150fec37/issues/gh/cdcgov/prime-reportstream/8621